cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterUserUpdateController.java

```diff
@@ -0,0 +1,110 @@
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import cwms.cda.data.dao.watersupply.WaterContractDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.watersupply.WaterUser;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+
+public final class WaterUserUpdateController extends WaterSupplyControllerBase implements Handler {
+
+    public WaterUserUpdateController(MetricRegistry metrics) {
+        waterMetrics(metrics);
+    }
+
+    @OpenApi(
+        requestBody = @OpenApiRequestBody(
+            content = {
+                @OpenApiContent(from = WaterUser.class, type = Formats.JSONV1)
+            },
+            required = true),
+        queryParams = {
+            @OpenApiParam(name = NAME, description = "Specifies the"
+                    + " new name of the water user entity.", required = true),
+        },
+        pathParams = {
+            @OpenApiParam(name = OFFICE, description = "The office Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The project Id the contract is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The water user the contract is associated with.",
+                    required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_204, description = "Water user successfully updated in CWMS."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not implemented")
+        },
+        description = "Updates a water user in CWMS.",
+        method = HttpMethod.PATCH,
+        path = "/projects/{office}/{project-id}/water-user/{water-user}",
+        tags = {TAG}
+    )
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(UPDATE)) {
+            DSLContext dsl = getDslContext(ctx);
+            String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) : Formats.JSONV1;
+            ContentType contentType = Formats.parseHeader(formatHeader, WaterUser.class);
+            ctx.contentType(contentType.toString());
+            String newName = ctx.queryParam(NAME);
```

Comment:
This should be a required param

Reply:
Added non null requirement, already marked as required in Swagger docs